pkg/specgen/container_validate.go

```diff
@@ -37,6 +37,23 @@ func (s *SpecGenerator) Validate() error {
 		}
 	}
 
+	// Containers being added to a pod cannot have certain network attributes
+	// associated with them because those should be on the infra container.
+	if len(s.Pod) > 0 {
```

Comment:
Suggest adding `&& s.NetNS.Mode == FromPod` (verifies we are not a `--net=host` container among other things)